README.md

```diff
@@ -10,6 +10,8 @@ Use this GitHub Action to [log in to a private container registry](https://docs.
 ```
 Refer to the action metadata file for details about all the inputs: [action.yml](https://github.com/Azure/docker-login/blob/master/action.yml)
 
+When using the docker-login action, ensure your login-server matches the fully qualified path to your image. You should omit login-server if you are pushing to the default docker hub without a host prefix, for instance, `docker push repo/image`. You'll need to specify a login-server if you are using the fully qualified path: `docker push index.docker.io/repo/image`. 

```

Comment:
docker defaults to `index.docker.io` if you don't specify a login server. Did you run into any specific issues?

Reply:
This holds true for all the other private registries though

Reply:
I ran into this: https://github.com/Azure/docker-login/issues/7#issuecomment-566407190 and wanted be explicit that when you don't prepend a hostname in your push/pull commands, you must omit login-server. Adding `login-server: index.docker.io` and then a `push or pull myrepo/myimage` fails with an auth issue.

Reply:
For push and pull you don't need to specify `index.docker.io` cause that's what `docker` defaults to. 

But in the example if you wrote something about a private registry like `acr` or something, I would have totally agreed because that's not the default behaviour.

Reply:
Right. What wasn't clear to me, or the person who posted #7, was that it will if you specify index.docker.io in one place and not other. 

Just trying to bring clarity here in the readme. It sounds like you might prefer a rewrite of this specific problem, or don't feel it warrants an update?

Reply:
I totally agree that the readme could use some additional clarity.

I somehow didn't feel that this statement was delivering what was expected. Specially because in this case the default `login-server`'s input value is `https://index.docker.io/v1` and not just `index.docker.io`.

You could probably add specific sample workflow yaml snippets maybe? to make it clear and differentiate between the default behavior vs the manual entry of login-server behaviour. 

And when you choose the default docker hub as your registry, you don't need to specify `index.docker.io/my-repo/my-image` just using `my-repo/my-image` would work for your `docker build, push` commands. Something like `docker build . -t my-repo/my-image`

But when you choose a different registry like `myacr.azurecr.io`; then you have to explicitly specify the login-server value in your docker build and push commands. You'll have to use it like `docker build . -t myacr.azurecr.io/my-image-name:version`